site/src/pages/UsersPage/UsersPage.test.tsx

```diff
@@ -48,53 +47,51 @@ const suspendUser = async (setupActionSpies: () => void) => {
   setupActionSpies()
 
   // Click on the "Confirm" button
-  const confirmButton = within(confirmDialog).getByText(UsersPageLanguage.suspendDialogAction)
-  fireEvent.click(confirmButton)
+  const confirmButton = await within(confirmDialog).findByText(
+    UsersPageLanguage.suspendDialogAction,
+  )
+  await user.click(confirmButton)
 }
 
 const deleteUser = async (setupActionSpies: () => void) => {
-  // Get the first user in the table
-  const users = await screen.findAllByText(/.*@coder.com/)
-  const firstUserRow = users[0].closest("tr")
-  if (!firstUserRow) {
-    throw new Error("Error on get the first user row")
-  }
+  const user = userEvent.setup()
+  // Click on the "more" button to display the "Delete" option
+  // Needs to await fetching users and fetching permissions, because they're needed to see the more button
+  const moreButtons = await screen.findAllByLabelText("more")
+  // get MockUser2
+  const selectedMoreButton = moreButtons[1]
 
-  // Click on the "more" button to display the "Suspend" option
-  const moreButton = within(firstUserRow).getByLabelText("more")
-
-  fireEvent.click(moreButton)
+  await user.click(selectedMoreButton)
 
   const menu = await screen.findByRole("menu")
-  const suspendButton = within(menu).getByText(UsersTableBodyLanguage.deleteMenuItem)
+  const deleteButton = within(menu).getByText(UsersTableBodyLanguage.deleteMenuItem)
 
-  fireEvent.click(suspendButton)
+  await user.click(deleteButton)
 
   // Check if the confirm message is displayed
   const confirmDialog = await screen.findByRole("dialog")
   expect(confirmDialog).toHaveTextContent(
-    `${UsersPageLanguage.deleteDialogMessagePrefix} ${MockUser.username}?`,
+    t("deleteDialog.confirm", { ns: "common", entity: "user" }),
   )
 
+  // Confirm with text input
+  const labelText = t("deleteDialog.confirmLabel", { ns: "common", entity: "user" })
+  const textField = screen.getByLabelText(labelText)
+  const dialog = screen.getByRole("dialog")
+  await user.type(textField, MockUser2.username)
+
   // Setup spies to check the actions after
   setupActionSpies()
 
   // Click on the "Confirm" button
-  const confirmButton = within(confirmDialog).getByText(UsersPageLanguage.deleteDialogAction)
-  fireEvent.click(confirmButton)
+  const confirmButton = within(dialog).getByRole("button", { name: "Delete" })
+  await user.click(confirmButton)
 }
 
 const activateUser = async (setupActionSpies: () => void) => {
-  // Get the first user in the table
-  const users = await screen.findAllByText(/.*@coder.com/)
-  const firstUserRow = users[2].closest("tr")
-  if (!firstUserRow) {
-    throw new Error("Error on get the first user row")
-  }
-
-  // Click on the "more" button to display the "Activate" option
-  const moreButton = within(firstUserRow).getByLabelText("more")
-  fireEvent.click(moreButton)
+  const moreButtons = await screen.findAllByLabelText("more")
+  const suspendedMoreButton = moreButtons[2]
+  fireEvent.click(suspendedMoreButton)
```

Comment:
Curious why we're using a `fireEvent` instead of an `await user.click` like above.

Reply:
It should be fine to do so - I just didn't make as many changes in this test. I wanted to switch to getting the "more" buttons everywhere because they load later than users, but the `userEvent` change was more of a hygiene thing than a cause of bugs.

Reply:
Got it! Thanks